### ISSUE TITLE
Explain why zero the gradient buffers of all parameters and backprops with random gradients

### ISSUE THREAD

**Issue by jlin27** [pytorch/tutorials#781]
In the [Neural Networks tutorial](https://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html#sphx-glr-beginner-blitz-neural-networks-tutorial-py), we zero the gradient buffers of all parameters and backprops with random gradients. Explain why are we doing this. 

![image](https://user-images.githubusercontent.com/8042156/70948708-e6162d00-2010-11ea-80ba-79c8b1b24b6e.png)

**Comment by parmarsuraj99** [comment]
I think it's because we haven't calculated the **loss** yet, So just to explain the **backward()**, it is done with random values. It's called on the outputs not the loss.

Gradients are calculated later from the loss. `loss.backward()`

**Comment by jlin27** [comment]
Addressed in PR #922
